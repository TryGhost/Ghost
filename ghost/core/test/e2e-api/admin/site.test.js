const {agentProvider, matchers} = require('../../utils/e2e-framework');
const {anyEtag, stringMatching, anyContentLength, anyContentVersion} = matchers;

describe('Site API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
    });

    it('can retrieve config and all expected properties', async function () {
        await agent
            .get('site/')
            .matchBodySnapshot({
                site: {
                    version: stringMatching(/\d+\.\d+/)
                }
            })
            .matchHeaderSnapshot({
                etag: anyEtag,
                'content-length': anyContentLength,
                'content-version': anyContentVersion
            });
    });
});
