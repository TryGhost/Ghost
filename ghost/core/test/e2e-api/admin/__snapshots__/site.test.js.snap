// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site API can retrieve config and all expected properties 1: [body] 1`] = `
Object {
  "site": Object {
    "accent_color": "#FF1A75",
    "allow_external_signup": true,
    "cover_image": "https://static.ghost.org/v5.0.0/images/publication-cover.jpg",
    "description": "Thoughts, stories and ideas",
    "icon": null,
    "locale": "en",
    "logo": null,
    "title": "Ghost",
    "url": "http://127.0.0.1:2369/",
    "version": StringMatching /\\\\d\\+\\\\\\.\\\\d\\+/,
  },
}
`;

exports[`Site API can retrieve config and all expected properties 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": StringMatching /\\\\d\\+/,
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
